src/main/java/org/broadinstitute/barclay/help/BashTabCompletionDoclet.java

```diff
@@ -0,0 +1,533 @@
+            final List<Map<String, String>> featureMaps) {
+
+            // For the Bash Test Doclet, this is a no-op.
+            // We only care about the index file.
+    }
+
+    /**
+     * The Index file in the Bash Test Doclet is what generates the actual tab-completion script.
+     *
+     * This will actually write out the shell completion output file.
+     * The Freemarker instance will see a top-level map that has two keys in it.
+     *
+     * The first key is for caller script options:
+     *
+     * SimpleMap callerScriptOptions = SimpleMap {
+     *
+     *   "callerScriptName"                 : caller script name
+     *
+     *   "callerScriptPrefixLegalArgs"      : caller Script Prefix Legal Args
+     *   "callerScriptPrefixArgValueTypes"  : caller Script Prefix Arg Value Types
+     *   "callerScriptPrefixMutexArgs"      : caller Script Prefix Mutex Args
+     *   "callerScriptPrefixAliasArgs"      : caller Script Prefix Alias Args
+     *   "callerScriptPrefixMinOccurrences" : caller Script Prefix Min Occurrences
+     *   "callerScriptPrefixMaxOccurrences" : caller Script Prefix Max Occurrences
+     *   "hasCallerScriptPrefixArgs"        : has Caller Script Prefix Args
+     *
+     *   "callerScriptPostfixLegalArgs"      : caller Script Postfix Legal Args
+     *   "callerScriptPostfixArgValueTypes"  : caller Script Postfix Arg Value Types
+     *   "callerScriptPostfixMutexArgs"      : caller Script Postfix Mutex Args
+     *   "callerScriptPostfixAliasArgs"      : caller Script Postfix Alias Args
+     *   "callerScriptPostfixMinOccurrences" : caller Script Postfix Min Occurrences
+     *   "callerScriptPostfixMaxOccurrences" : caller Script Postfix Max Occurrences
+     *   "hasCallerScriptPostfixArgs"        : has Caller Script Postfix Args
+     *
+     * }
+     *
+     * The second key is for tool options:
+     *
+     * SimpleMap tools = SimpleMap { ToolName : MasterPropertiesMap }
+     *
+     *     where
+     *
+     *     MasterPropertiesMap is a map containing the following Keys:
+     *         all
+     *         common
+     *         positional
+     *         hidden
+     *         advanced
+     *         deprecated
+     *         optional
+     *         dependent
+     *         required
+     *
+     *         Each of those keys maps to a List&lt;SimpleMap&gt; representing each property.
+     *         These property maps each contain the following keys:
+     *
+     *             kind
+     *             name
+     *             summary
+     *             fulltext
+     *             otherArgumentRequired
+     *             synonyms
+     *             exclusiveOf
+     *             type
+     *             options
+     *             attributes
+     *             required
+     *             minRecValue
+     *             maxRecValue
+     *             minValue
+     *             maxValue
+     *             defaultValue
```

Comment:
Cool.  Added in minElements and maxElements.

This is slightly different in that it also has the callerScriptOptions map, but yeah - the tools map is what is generated by DefaultDocWorkUnitHandler.  @cmnbroad do you want me to move this into there?  If so, I can append it to the end of the comment on processWorkUnit.